orc/src/main/java/org/apache/iceberg/data/orc/GenericOrcWriters.java

```diff
@@ -531,6 +543,34 @@ public void nonNullWrite(int rowId, Map<K, V> map, ColumnVector output) {
     }
   }
 
+  private static class DeleteWriter implements OrcRowWriter<PositionDelete<Record>> {
+    private final OrcRowWriter<Record> rowWriter;
+    private final Schema deleteSchema;
+    private final GenericRecord record;
+
+    DeleteWriter(Schema deleteSchema, OrcRowWriter rowWriter) {
+      this.deleteSchema = deleteSchema;
```

Comment:
Do we need to pass `deleteSchema` here?
I think it is only used to construct `rowWriter` that we can build in `ORC#buildPositionDeleteWriter`.

Reply:
We need the `deleteSchema` to create the record which we pass to the writer:
```
this.record = GenericRecord.create(deleteSchema);
```


About constructing the writer in `ORC#buildPositionDeleteWriter`:
- I think the creation of the internal writer is the responsibility of the `PositionDeleteWriter` so if possible it should be done here
- Since we allow `createWriterFunc` which gives a way to the user to provide its own writer implementation, we should find a way to accept a writer which could be used instead of the default one when writing

If you agree with the things above I might add a new method to make it more clear:
```
  public static PositionDeleteWriter positionDelete(Schema schema) {
    return new PositionDeleteWriter(schema, null);
  }
```
What do you think?